tests/pds/dataservice_test.go

```diff
@@ -1666,3 +1666,67 @@ var _ = Describe("{AddAndValidateUserRoles}", func() {
 		EndTorpedoTest()
 	})
 })
+var _ = Describe("{GetPvcToFullCondition}", func() {
+
+	JustBeforeEach(func() {
+		InitInstance()
+		StartTorpedoTest("GetPvcToFullCondition", "Deploys and increases the pvc size of DS once trhreshold is met", pdsLabels, 0)
+	})
+
+	It("Deploy Dataservices", func() {
+		var generateWorkloads = make(map[string]string)
+		var deployments = make(map[PDSDataService]*pds.ModelsDeployment)
+		var dsVersions = make(map[string]map[string][]string)
+		var depList []*pds.ModelsDeployment
+		var dsName string
+
+		Step("Deploy Data Services", func() {
+			for _, ds := range params.DataServiceToTest {
+				Step("Deploy and validate data service", func() {
+					isDeploymentsDeleted = false
+					controlPlane.UpdateResourceTemplateName("pds-auto-pvcFullCondition")
+					deployment, _, dataServiceVersionBuildMap, err = DeployandValidateDataServices(ds, params.InfraToTest.Namespace, tenantID, projectID)
+					log.FailOnError(err, "Error while deploying data services")
+					deployments[ds] = deployment
+					dsVersions[ds.Name] = dataServiceVersionBuildMap
+					depList = append(depList, deployment)
+					dsName = ds.Name
+
+				})
+			}
+			// This testcase is currently applicable only for postgresql deployments
+			if dsName == postgresql {
+				Step("Running Workloads before scaling up PVC ", func() {
+					for ds, deployment := range deployments {
+						if Contains(dataServicePodWorkloads, ds.Name) || Contains(dataServiceDeploymentWorkloads, ds.Name) {
+							log.InfoD("Running Workloads on DataService %v ", ds.Name)
+							var params pdslib.WorkloadGenerationParams
+							pod, dep, err = RunWorkloads(params, ds, deployment, namespace)
+							log.FailOnError(err, fmt.Sprintf("Error while genearating workloads for dataservice [%s]", ds.Name))
+							if dep == nil {
+								generateWorkloads[ds.Name] = pod.Name
+							} else {
+								generateWorkloads[ds.Name] = dep.Name
+							}
+							for dsName, workloadContainer := range generateWorkloads {
+								log.Debugf("dsName %s, workloadContainer %s", dsName, workloadContainer)
+							}
+						}
+					}
+				})
+
```

Comment:
Add defer function to clean up the created workload pods

Reply:
Done !